processor/src/main/java/org/mapstruct/ap/version/VersionInformation.java

```diff
@@ -0,0 +1,32 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.version;
+
+/**
+ * Provides information about the processor version and the processor context implementation version
+ *
+ * @author Andreas Gudian
+ */
+public interface VersionInformation {
+    String getRuntimeVersion();
+    String getRuntimeVendor();
+    String getMapStructVersion();
+    String getProcessingEnvironmentImplementationName();
```

Comment:
I'm wondering whether these last two could be condensed into a more high-level representation, e.g. "javac 1.7.0_45" or "ejc x.y.z"? The environment class name seems a bit too much to me. WDYT?

How about structuring the comment like so:

    "version: 1.0.0.Beta3, compiler: javac 1.7.0_45, environment: Java 1.8.0_20 (Oracle Corporation)"

?